### PR TITLE
Keychain

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes RNBW-1887
Fixes RNBW-1888
Fixes RNBW-1960

## What changed (plus any additional context for devs)
- Upgraded `react-native-keychain` to v8
- Upgraded `@segment/analytics-react-native` to v1.5 (otherwise it was crashing the app on android)
- Changed the way we store public items in the keychain (previous method was deprecated by apple)
- Added migration to update all existing public keychain items with the new method
- Moved migrations file to typescript
- Disabled the initial crypto "warm up" on android that slows down startup

## PoW (screenshots / screen recordings)

On a pixel 4 APK (release mode) initial launch is ~2x faster
probably much more noticeable in older devices.

Before: http://recordit.co/QqpSkuWXdq (~ 3 sec)
After  http://recordit.co/2x5n9phOCs (~1.5 sec)

## Dev checklist for QA: what to test
- Requires full regression

## Final checklist
[x] Assigned individual reviewers?
[x] Added labels?

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1063,4 +1063,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: ba8425d4154c1535ff42c6c92d19ca2be6f86110
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.10.2
```
Comment (jinchung) [inline ios/Podfile.lock:1066]: only thing is this went down

Comment (brunobar79) [inline ios/Podfile.lock:1066]:
```suggestion
COCOAPODS: 1.11.2
```

**Review by jinchung — APPROVED** [review]
🍔 🍟

**Comment by osdnk** [comment]
Did you change anything significant in the patch?

**Comment by brunobar79** [comment]
> Did you change anything significant in the patch?

No, just readded the same methods from before, on top of the new version

**Review by skylarbarrera — APPROVED** [review]
🐐

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good on both OS's, QA Passed 👍🏽

**Pull request merged**